lib/ClangImporter/ImportDecl.cpp

```diff
@@ -2189,12 +2189,24 @@ namespace {
 
               Impl.markUnavailable(MD, "use .pointee property");
               MD->overwriteAccess(AccessLevel::Private);
+            } else if (cxxOperatorKind ==
```

Comment:
I believe a class Foo could define `int operator++()` method.

Meaning (and I'm not sure why you'd do this) you could have the increment op return a diff type and it would then not fit into the mold of a "successor", "predecessor".

Maybe checking if the operator is ++ _and_ that the return type is equal to the class itself or something?

Reply:
Yeap, that's unfortunately valid C++.
This is why I decided to do this:
```swift
var __copy: Self
__copy = self
__copy.__operatorPlusPlus()
return __copy
```
instead of this:
```swift
var __copy: Self
__copy = self
return __copy.__operatorPlusPlus().pointee
```

However, I think you're right and it's better to disable such operators altogether. If there's a valid use case for them, we can lift the restriction later.